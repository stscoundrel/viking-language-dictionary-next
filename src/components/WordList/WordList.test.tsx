import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import WordList from './index'

const words = [
  {
    headword: 'þess',
    definition: 'm/n gen sg of sá/þat',
    isCommon: false,
    slug: 'thess',
  },
  {
    headword: 'þessi',
    definition: '(also sjá) <f þessi, n þetta, m acc sg þenna, m dat sg & all dat pl þessum~þeima, m/n gen sg þessa, m nom pl þessir, all gen pl þessa~þessar(r)a, f acc sg þessa, f dat sg þessi~þessar(r)i, f gen sg þessar~þessar(r)ar, n dat sg þessu~þvísa, n nom/acc pl þessi> dem pron this, these',
    isCommon: true,
    slug: 'thessi',
  },
  {
    headword: 'þetta',
    definition: 'n nom/acc sg of þessi',
    isCommon: false,
    slug: 'thetta',
  },
  {
    headword: 'þér',
    definition: '<acc/dat yðr, gen yðarr~yðvarr> pl pron you; þér frequent dat of þú; see ér',
    isCommon: false,
    slug: 'ther',
  },
  {
    headword: 'Þiðrandi',
    definition: 'm Thidrandi (personal name)',
    isCommon: false,
    slug: 'thidrandi',
  },
  {
    headword: 'þiggja',
    definition: '<þiggr, þá, þágu, þeginn> vb (V) to accept; receive; accept lodgings',
    isCommon: false,
    slug: 'thiggja',
  },
  {
    headword: 'þinglogi',
    definition: 'adj indecl engagement breaker',
    isCommon: false,
    slug: 'thinglogi',
  },
]

describe('WordList component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<WordList words={words} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<WordList words={words} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Renders correct amount of words', () => {
    const tree = renderer.create(<WordList words={words} />)
    const { root } = tree

    expect(root.findAllByType('ul').length).toEqual(1)
    expect(root.findAllByType('li').length).toEqual(7)
  })
})
