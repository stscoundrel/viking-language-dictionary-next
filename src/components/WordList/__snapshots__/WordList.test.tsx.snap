// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordList component Matches snapshot 1`] = `
<ul
  className="list"
>
  <li>
    <a
      className="link"
      href="/word/thess"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      þess
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/thessi"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      þessi
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/thetta"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      þetta
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/ther"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      þér
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/thidrandi"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      þiðrandi
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/thiggja"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      þiggja
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/thinglogi"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      þinglogi
    </a>
  </li>
</ul>
`;
